examples/components/utils/data_analysis/run.py

```diff
@@ -0,0 +1,240 @@
+
+    Args:
+        parser (argparse.ArgumentParser or CompliantArgumentParser):
+        an argument parser instance
+
+    Returns:
+        ArgumentParser: the argument parser instance
+
+    Notes:
+        if parser is None, creates a new parser instance
+    """
+    # add arguments that are specific to the component
+    if parser is None:
+        parser = argparse.ArgumentParser(description=__doc__)
+
+    parser.add_argument(
+        "--training_data", type=str, required=True, help="Training CSV file"
+    )
+    parser.add_argument("--testing_data", type=str, required=True, help="Test CSV file")
+    parser.add_argument(
+        "--categorical_columns",
+        type=str,
+        required=False,
+        help="Categorical column names separated by comma(,)",
+    )
+    parser.add_argument(
+        "--onehot_columns_prefix",
+        type=str,
+        required=False,
+        help="One hot encoded columns prefixes separated by comma(,), prefixes cannot contain underscores in the middle and must end with underscore",
+    )
+    parser.add_argument(
+        "--metrics_prefix",
+        type=str,
+        required=False,
+        help="Metrics prefix",
+        default="default-prefix",
+    )
+    return parser
+
+
+def get_corr_fig(df):
+    fig, ax = plt.subplots()
+    fig.set_figwidth(min(30, df.shape[1] * 2))
+    corr = df.corr()
+    cax = ax.matshow(corr)
+    ax.set_xticks(range(len(corr.columns)))
+    ax.set_yticks(range(len(corr.columns)))
+    ax.set_xticklabels(corr.columns.to_list(), rotation=45)
+    ax.set_yticklabels(corr.columns.to_list())
+    fig.colorbar(cax)
+    fig.tight_layout()
+    return fig
+
+
+def analyze_categorical_column(df, column_name, metrics_prefix):
+    value_counts = {
+        str(k): str(v) for k, v in dict(df[column_name].value_counts()).items()
+    }
+    mlflow.log_dict(value_counts, f"{metrics_prefix}/{column_name}/value_counts.json")
+
+    # Having figure larger than 30 inches is undesirable
+    if len(value_counts) <= 60:
+        fig, ax = plt.subplots()
+        fig.set_figwidth(len(value_counts) / 2)
+        ax = df[column_name].value_counts().plot(kind="bar")
+        ax.set_xticklabels(list(value_counts), rotation=45)
+        mlflow.log_figure(fig, f"{metrics_prefix}/{column_name}/density.png")
+    else:
+        logger.info(
+            f"Skipping plotting due too many unique values for column: {column_name}"
+        )
+
+
+def analyze_numerical_column(df, column_name, metrics_prefix):
+    column_description = {
+        str(k): str(v) for k, v in dict(df[column_name].describe()).items()
+    }
+    mlflow.log_dict(
+        column_description, f"{metrics_prefix}/{column_name}/column_description.json"
+    )
+
+    fig, ax = plt.subplots()
+    ax = df[column_name].plot(
+        kind="density", xlim=(df[column_name].min(), df[column_name].max())
+    )
+    mlflow.log_figure(fig, f"{metrics_prefix}/{column_name}/density.png")
```

Comment:
can we record all those mlflow calls to the pipeline root level?

This is what is done in the training script, and avoids having users to navigate to each step to find the relevant metrics